src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj

```diff
@@ -95,11 +104,18 @@
       (onWebSocketClose [statusCode ^String reason]
         (let [^WebSocketAdapter this this]
           (proxy-super onWebSocketClose statusCode reason))
+        (.countDown closureLatch)
         (on-close this statusCode reason))
       (onWebSocketBinary [^bytes payload offset len]
         (let [^WebSocketAdapter this this]
           (proxy-super onWebSocketBinary payload offset len))
         (on-bytes this payload offset len))
+      (awaitClosure []
+        (try
+          (when-not (.await closureLatch 30 TimeUnit/SECONDS)
```

Comment:
Probably we don't want this 30 seconds hardcoded, thoughts on what the value should be and if it should be configurable?